### PR TITLE
use ipaddress module to determine loopback interfaces

### PR DESCRIPTION
**Pull request by mgfritch**
handling for other loopback address variants (127.0.1.1, ::1, etc.)

Signed-off-by: Michael Fritch <mfritch@suse.com>

### CODE REVIEW

File: ceph_salt/core.py [hunk 1]
```diff
@@ -44,18 +45,21 @@ def ipsv6(self):
 
     @property
     def public_ip(self):
+        def _is_loopback(addr):
+            return ipaddress.ip_address(addr).is_loopback
```
Comment (kshtsk) [inline ceph_salt/core.py:49]: Cool, this looks covers 127.0.1.1 too:
```
>>> ipaddress.ip_address('127.0.1.1').is_loopback
True
```

File: ceph_salt/core.py [hunk 2]
```diff
@@ -44,18 +45,21 @@ def ipsv6(self):
...
+            if _is_loopback(_public_ip):
+                logger.debug('fqdn_ipv4 grain is a loopback address, falling back to ipv4 grain')
                 result = GrainsManager.get_grain(self.minion_id, 'ipv4')
                 for addr in result[self.minion_id]:
-                    if addr != '127.0.0.1':
+                    if not _is_loopback(addr):
                         _public_ip = addr
                         break
-                if _public_ip == '127.0.0.1':
-                    logger.warning("'%s' public IP is the loopback interface IP ('127.0.0.1')",
```
Comment (kshtsk) [inline ceph_salt/core.py:58]: Would be great if you add _public_ip value to the log instead of 127.0.0.1

File: ceph_salt/core.py [hunk 3]
```diff
@@ -44,18 +45,21 @@ def ipsv6(self):
...
     @property
     def public_ip(self):
+        def _is_loopback(addr):
+            return ipaddress.ip_address(addr).is_loopback
+
         if self._public_ip is None:
             result = GrainsManager.get_grain(self.minion_id, 'fqdn_ip4')
             _public_ip = result[self.minion_id][0]
-            if _public_ip == '127.0.0.1':
-                logger.debug('fqdn_ipv4 grain is 127.0.0.1, falling back to ipv4 grain')
```
Comment (kshtsk) [inline ceph_salt/core.py:51]: For better debugging I guess it is better to print the _public_ip here the same way as it was before

**Review by kshtsk — APPROVED** [review]
Looks good, but minor debugging/logging improvements required

File: ceph_salt/core.py (hunk 2 above)
Comment (mgfritch) [reply ceph_salt/core.py:58]: yeah good point, I've re-added the `_public_ip` to the log message

File: ceph_salt/core.py (hunk 3 above)
Comment (mgfritch) [reply ceph_salt/core.py:51]: added `_public_ip` as part of the log message

**Review by tserong — APPROVED** [review]
Nicely done

**Pull request merged**